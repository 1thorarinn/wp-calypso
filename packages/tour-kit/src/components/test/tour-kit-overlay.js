/**
 * @jest-environment jsdom
 */
import { shallow } from 'enzyme';
import TourKitOverlay from '../tour-kit-overlay';

describe( 'TourKitOverlay', () => {
	test( 'should render', () => {
		const wrapper = shallow( <TourKitOverlay /> );

		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.type() ).toBe( 'div' );
	} );

	test( 'should render only with class tour-kit-overlay when visible var not passed', () => {
		const wrapper = shallow( <TourKitOverlay /> );

		expect( wrapper.hasClass( 'tour-kit-overlay' ) ).toBe( true );
		expect( wrapper.hasClass( '--visible' ) ).toBe( false );
	} );

	test( 'should render only with class tour-kit-overlay if visible var is false', () => {
		const wrapper = shallow( <TourKitOverlay visible={ false } /> );

		expect( wrapper.hasClass( 'tour-kit-overlay' ) ).toBe( true );
		expect( wrapper.hasClass( '--visible' ) ).toBe( false );
	} );

	test( 'should render also with class visible if visible var is true', () => {
		const wrapper = shallow( <TourKitOverlay visible={ true } /> );

		expect( wrapper.hasClass( 'tour-kit-overlay' ) ).toBe( true );
		expect( wrapper.hasClass( '--visible' ) ).toBe( true );
	} );
} );
