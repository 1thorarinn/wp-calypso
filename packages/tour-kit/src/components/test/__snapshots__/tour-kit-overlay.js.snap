// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourKitOverlay should render 1`] = `
<div
  className="tour-kit-overlay"
/>
`;
